spark_board/plan_extractor/transformation_node_builders.py

```diff
@@ -377,7 +377,25 @@ def _expected_number_of_nodes(self) -> Optional[int]:
         return 1
 
 
-def build_dag_from_java_object(node: JavaObject) -> TransformationNode:
+class UnknownNodeBuilder(TransformationNodeBuilder):
+    """This class is used to build nodes for transformations that are not supported.
+    Instead of failing and leaving the users without any visualization, we can opt
+    to return an "unknown" transformation and at least provide some kind of output.
+    """
+
+    def _extract_metadata(self, node: JavaObject, metadata: Metadata) -> None:
+        metadata["inferred_type"] = node.getClass().getSimpleName()
+        # toString() is a method of the `TreeNode` class, so it should be available in all nodes
```

Comment:
actually all Java Objects have toString()

Reply:
Yes, but the point is that it makes sense to use it as default representation (although idk if there's a better option).